sphinx/builders/changes.py

```diff
@@ -127,7 +127,7 @@ def hl(no: int, line: str) -> str:
             with open(targetfn, 'w', encoding='utf-8') as f:
                 text = ''.join(hl(i + 1, line) for (i, line) in enumerate(lines))
                 ctx = {
-                    'filename': self.env.doc2path(docname, None),
+                    'filename': self.env.doc2path(docname, False),
```

Comment:
Why is the type for `filename` `str | Literal[False]`? `str | None` would make more sense, or just the empty string here if we're wanting a falsy value.

Reply:
i'm not sure I understand the question.

> Why is the type for `filename` `str | Literal[False]`

where are you getting that from?

Reply:
My mistake, thanks. I got mixed up and was thinking of the second argument as the default/fallback.

A